src/Compiler/ComponentTagCompiler.php

```diff
@@ -236,4 +241,27 @@ protected function parseAttributeBag(string $attributeString): string
 
         return (string) preg_replace($pattern, ' :attributes="$1"', $attributeString);
     }
+
+    /**
+     * Remove Twig comments from given attribute string
+     */
+    protected function pourOutCommentsFromAttributeBag(string $attributeString): string
+    {
+        $pattern = "/
+            (                                               # First capturing group
+                \{\#                                        # Match {# literally
+                    (                                       # Second capturing group
+                        [                                   # Match any character in the following list
+                            \w                              # Match words
+                            \d                              # Match digits
+                            \s                              # Match whitespace characters
+                            $&+,:;=?@#|\"'`<>.^*()%!-       # Match special characters
+                        ]+                                  # End of list
```

Comment:
@literat Try this:

```suggestion
                        (?!\#\}).+?                         # Use negative lookahead to match until the first occurrence of #}
```

but you also need to add the `/s` modifier (for multiline match). Or you need to add the multiline characters to the match group like this: `(?!\#\})(.|\s)+?`.

Unlike the original solution, this will match multilines and event the case mentioned in the [previous comment](https://github.com/lmc-eu/twigx-bundle/pull/19/files#r1081268729.

Reply:
Great solution. Fixes everything :-) 👍 Thanks.